### PR TITLE
Church file split, s4a - Church first visit, and s4c - Church rap battle

### PR DESCRIPTION
**Pull request by catdad666**
I split the Church files into four sections so that it's easier to tackle. I also drafted the first visit to the Church, which should slot in alongside the Roach Market and Gutter.

I changed the efjm.tw2 file to include all of the new file names and added a debug link for the first Church visit scene.

Nothing fancy in the first Church visit scene, but feel free to play through and enjoy Ron's letter. I formatted it like the letters from Anna as a fun joke.

### CODE REVIEW

**Comment by catdad666** [comment]
Added the rap battle to this PR. It also includes the end scene. Didn't put as many transitions in as we'll probably have in the final game. But, it's functional for now.

File: src/efjm.tw2 [hunk 1]
```diff
@@ -28,9 +30,30 @@ src/s2a_moms_tent.tw2
...
+Ben Morgan
+
+**Art by**
+Sarah Morgan
+
+**Special Thanks**
+Meemaw and Pep Pep
+
+**Playtesters**
+Violent J
```
Comment (whymog) [inline src/efjm.tw2:58]: Would that we were so lucky.

File: src/s4a_the_church_first_visit.tw2 [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+<div style="padding-left: 2em;">//Dear Ron,
+
+This is also Ron hehe what’s up good buddy?? :) Anyway, don’t forget to come unlock the door 2nite so that fukkin’ straglers can come in to the chalenge the champs 2nite.
+
+(click: "champs 2nite")[$dissolve[I no u wont buddy but the boss said rite a note. Keep this on u at all times!!!]]
+
+(click: "at all times")[$dissolve[EVEN WHEN U GO TO BAR!! Prove to the boss ug ot this one man!!
+
+Love,
+[Ron]<finalRon|]]//</div>
```
Comment (whymog) [inline src/s4a_the_church_first_visit.tw2:103]: THE FINAL RON

File: src/s4a_the_church_first_visit.tw2 [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+(click: ?finalRon)[$dissolve[On the other side of the note Ron had written a bunch of tally marks under the words “WERK BEERZ”, I guess in an attempt to judge when he’d had too many.
```
Comment (whymog) [inline src/s4a_the_church_first_visit.tw2]: The comma after "WERK BEERZ" looks a little odd to me. How about this:

>On the other side of the note, Ron had written a bunch of tally marks under the words “WERK BEERZ”. I guessed this was his attempt to judge when he’d had too many.

File: src/s4a_the_church_first_visit.tw2 [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+There were 10+ tally marks and the words “UH OH” kind of scribbled afterwards.
```
Comment (whymog) [inline src/s4a_the_church_first_visit.tw2]: "10+" reads weirdly to me. How about:

>There were at least 10 tally marks on the sheet, and the words "UH OH" were kind of scribbled afterwards.

File: src/s4c_the_church_rap_battle.tw2 [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+As I moved to pick my mic up from where it was coiled on the stage floor, some stupid-ass juggalo in the crowd called out, “Yo, I ain’t ever seen this bitch. She ain’t got it!” Steve popped a mile-wide grin and there was some laughter in the crowd.]]
+
+(click: "laughter in the crowd")[$dissolve[I quickly snatched up the mic and blasted back, “Yeah ninja, that ain’t a surprise. Looks to me like a lot of ‘lettes try to avoid you all together.” An immediate “OOOOOOH” from the crowd was my delicious reward. Steve’s grin dropped and he picked his mic up as well. [[Time to do some fuckin' raps|The Church - Rap battle - First verses]].]]
+
+::The Church - Rap battle - First verses
+Steve and I circled each other warily. Moon Mystic piped up, “Alright $stageName, as the challenger, you pick who goes first.”
+
+>>["I’ll go first. I’m sick of looking at this dude already."]<playerFirst|
+>>["I’ll go second. This goofy motherfucker needs all the help he can get."]<steveFirst|
+>>[[“I didn’t even answer, just started into a blistering rap about dogs.”|The Church - Rap battle - First verses dog rap]]
```
Comment (whymog) [inline src/s4c_the_church_rap_battle.tw2]: This option shouldn't be in quotes, right?

Comment (whymog) [reply src/s4c_the_church_rap_battle.tw2]: Also, I'd swap the comma for an em-dash

File: src/s4c_the_church_rap_battle.tw2 [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+
+[(live: 1s)[(if: time > 17s)[$dissolve[(if: $steveRapsFirst is true)[I could feel the power coursing through me and it matched the audience’s roar of approval. [[What a perfect time for the results|The Church - Rap battle - Victory player second]].]
+\(else:)[I could feel the power coursing through me and it matched the audience’s roar of approval. [[Time for Steve’s verse. Ugh|The Church - Rap battle - Second verses steve]].]
+(stop:)]]]]
+
+::The Church - Rap battle - Second verses dogs
+I had to respond to what was in my heart, and what was moving the crowd, so I kept rapping about dogs.
+
+[(live: 1s)[(if: time > 1s)[$dissolve[//I wish I had a puppy when I was growing up
+I’d love to have a flopsy, friendly lil mutt//(stop:)]]]]
```
Comment (whymog) [inline src/s4c_the_church_rap_battle.tw2]: Should be "li'l"

File: src/s4c_the_church_rap_battle.tw2 [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+[(live: 1s)[(if: time > 3s)[$dissolve[//It’d be huge now, with an empty gut
+Waiting for my command to eat Steve’s nuts//(stop:)]]]]
+
+[(live: 1s)[(if: time > 5s)[$dissolve[//Yo I’m wild for any dog, especially the shih-tzu’s
+Dogs look at Steve like, “You shit outside too dude?”//(stop:)]]]]
```
Comment (whymog) [inline src/s4c_the_church_rap_battle.tw2]: Let's add a comma like:

>"You shit outside too, dude?"

File: src/s4c_the_church_rap_battle.tw2 [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+
+We hung backstage for a while, recapping the greatest hits of the match and talking strategy, until I got called to the stage for [[the ceremony|The Church - Mountain ceremony]]. As Kiki squeezed me one last time, she said “Good luck on the mountain, Double L. You’re gonna be amazing.” Then I was gone, but her strength still flowed through me like pure caffeine.
+
+::The Church - Mountain ceremony
+The spotlight hit me as Moon Mystic’s assistants, decked in traditional neon green and black robes, lifted me up in an aluminum lawn chair. I bathed in the light and the crowd’s adoration as Moon Mystic read the rites over an open bottle of his namesake.
+
+(click: "an open bottle of his namesake")[$dissolve[“Take and drink. This is that good shit. May the haters take and eat unclean ass. You are the champ, and a ninja, and the most bomb-ass motherfucker forever and ever amen.” The crowd responded with a hearty amen. Moon Mystic dumped a beige powder into the bottle of Faygo and let it fizz a bit before offering it up to me.]]
```
Comment (whymog) [inline src/s4c_the_church_rap_battle.tw2]: Let's add a comma after "forever and ever"

File: src/s4c_the_church_rap_battle.tw2 [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+[(live: 1s)[(if: time > 17s)[$dissolve[Some of Steve’s boys made loud noises, but the dogs were curiously silent after Steve’s verse. We [[moved into round two|The Church - Rap battle - Second verses dogs]].
+(stop:)]]]]
+
+::The Church - Rap battle - Second verses steve
+Steve blubbered and wavered during his second verse like a jellyfish having an asthma attack.
+
+[(live: 1s)[(if: time > 1s)[$dissolve[//Yo I’m Steve here to tear this shit apart
+I sneak in your front door like a fart//(stop:)]]]]
+[(live: 1s)[(if: time > 3s)[$dissolve[//You ain’t want me, but you need me
+Everyday I’m all over that shit baby//(stop:)]]]]
```
Comment (whymog) [inline src/s4c_the_church_rap_battle.tw2]: "Everyday" => "Every day"

File: src/s4c_the_church_rap_battle.tw2 [hunk 10]
```diff
@@ -0,0 +1,343 @@
...
+Shout out to the dachshunds and all the little homies
+Shoutin’ shut the fuck up Steve, you don’t know me//(stop:)]]]]
+
+[(live: 1s)[(if: time > 17s)[$dissolve[The dog crowd had been standing during the whole verse and now they were rushing towards the stage. Steve was looking pretty sweaty, but he said, “Fuck that weird shit. I’m doing my thing,” and [[launched into his next verse|The Church - Rap battle - Victory dog raps]].
+(stop:)]]]]
+
+::The Church - Rap battle - Victory player first
+It was clear that the crowd had been with me from the start. I could read the energy. As Steve’s homies filed out, a fired up group of ‘lettes stormed the stage and started chanting my name. Some of them were new faces, others I knew and had even heard talk shit. I didn’t care. All were welcome in my temple now that I had made it to the top.
+
+Moon Mystic approached Steve and I, holding both our hands, but I didn’t even wait for him. I raised my own hand and screamed, “YOUR CHAMPION [STAGE NAME]!” Moon Mystic laughed and nodded as [[the crowd stomped in unison|The Church - Mountain intro]].
```
Comment (whymog) [inline src/s4c_the_church_rap_battle.tw2]: Looks like this is supposed to be a variable? Also a few lines down, same thing.

File: src/s4c_the_church_rap_battle.tw2 [hunk 11]
```diff
@@ -0,0 +1,343 @@
...
+::The Church - Mountain ending choice
+I knew even then that no one had finished the mountain since the Honchos had put it in place. I wouldn’t really start caring until I got into that son of a bitch. Anna was waiting for me somewhere in there and nothing had stopped me yet.
+
+What I didn't know was that:
+>>[[this was the beginning of the longest night of my life. My last night in Gatherville.|EFJM - Credits]]
+>>[[Anna was waiting for me too — waiting to kill me.|EFJM - Credits]]
+>>[[Anna and I were already done. I would have to betray my best friend.|EFJM - Credits]]
+>>[[Ron was my real dad and later that night he’d cut my hand off. What a fucking piece of shit.|EFJM - Credits]]
+>>[[this story was constructed in the past tense so that the dumbass dude writing this could do a crazy reveal where I’m like the president now or some shit. Haha get fucked.|EFJM - Credits]]
+>>[[the world outside Gatherville was even worse. Jeff Bezos was the president and his powerful drones could only give you oral sex or cut off your head and they charged the same amount of money for either.|EFJM - Credits]]
```
Comment (whymog) [inline src/s4c_the_church_rap_battle.tw2]: I feel like each of these should lead to a "To be continued" kinda thing before the credits. What do you think?

**Comment by whymog** [comment]
Looks great! Just a few comments.

**Review by whymog — APPROVED** [review]

File: src/s4c_the_church_rap_battle.tw2 (hunk 11 above)
Comment (catdad666) [reply src/s4c_the_church_rap_battle.tw2]: Yeah, that works for me. I'll build something as a placeholder so we don't forget. Maybe something that displays "To Be Continued" centered and then displays the credits after a set period of time.

File: src/s4c_the_church_rap_battle.tw2 (hunk 10 above)
Comment (catdad666) [reply src/s4c_the_church_rap_battle.tw2]: LOL, this is what I get for inputting text too fast and then always picking the dog victory option when testing the ending. Thanks for catching this!

File: src/s4a_the_church_first_visit.tw2 (hunk 2 above)
Comment (catdad666) [reply src/s4a_the_church_first_visit.tw2:103]: Definitely had created enough variables at this point that I was starting to goof off a bit.

File: src/s4c_the_church_rap_battle.tw2 (hunk 5 above)
Comment (catdad666) [reply src/s4c_the_church_rap_battle.tw2]: whooooops

File: src/s4c_the_church_rap_battle.tw2 (hunk 6 above)
Comment (catdad666) [reply src/s4c_the_church_rap_battle.tw2]: Tried to look up the correct version of this and found the most odious spelling forum arguments of all time. I would fight for "Lil" with no apostrophes as the beginning of a rapper name, but in this context your version makes sense.

File: src/s4c_the_church_rap_battle.tw2 (hunk 8 above)
Comment (catdad666) [reply src/s4c_the_church_rap_battle.tw2]: Hmmm interesting. I'm used to saying this very quickly at a church with no pauses, but Randy Travis' song of the same name agrees with you. I added the comma.

**Pull request merged**